openpgp/keys.go

```diff
@@ -540,11 +541,13 @@ func (e *Entity) SerializePrivate(w io.Writer, config *packet.Config) (err error
 }
 
 // SerializePrivate serializes an Entity, including private key material, to
-// the given Writer. For now, it must only be used on an Entity returned from
-// NewEntity.
+// the given Writer.
 // If config is nil, sensible defaults will be used.
-// TODO::notes this is a temp function to avoid break other things
 func (e *Entity) SerializePrivateNoSign(w io.Writer, config *packet.Config) (err error) {
+	if e.PrivateKey == nil {
```

Comment:
@twiss Does this make sense to enforce?

Reply:
Yeah, I think this makes sense :+1: You could add it to `SerializePrivate` as well if you like

Reply:
Perfect, added :+1: